docs/src/manual/fe_intro.md

```diff
@@ -1,111 +1,184 @@
+of the boundary, $\Gamma_\mathrm{D}$, and the heat flux is prescribed to $q^\mathrm{p}$ at
+the Neumann part of the boundary, $\Gamma_\mathrm{N}$, where $\mathbf{n}$ describes the outward
+pointing normal vector at the boundary.
 
 We also need a constitutive equation which links the temperature field, $u$, to the heat
 flux, $\mathbf{q}$. The simplest case is to use Fourier's law
 
 ```math
-\mathbf{q} = -k \mathbf{\nabla}u
+\mathbf{q}(u) = -k \nabla u
 ```
 
 where $k$ is the conductivity of the material. For simplicity we will consider only
 constant conductivity $k$.
 
-## Weak format
+## Weak Form
 
 The solution to the equation above is usually calculated from the corresponding weak
-format. By multiplying the equation with an arbitrary test function $\delta u$, integrating
-over the domain and using partial integration we obtain the *weak form*;
+form. By multiplying the equation with an arbitrary *test function* $\delta u$, integrating
+over the domain and using partial integration we obtain the *weak form*. Now our problem
+can be stated as:
+
 Find $u \in \mathbb{U}$ s.t.
 
 ```math
-\int_\Omega \mathbf{\delta u} \cdot (k \mathbf{u}) \mathrm{d}\Omega =
-\int_{\Gamma_\mathrm{N}} \delta u q^\mathrm{p} \mathrm{d}\Gamma +
-\int_\Omega \delta u b \mathrm{d}\Omega \quad \forall \delta u \in \mathbb{U}^0
+\int_\Omega \nabla \delta u \cdot (k \nabla u) \, \mathrm{d}\Omega =
+\int_{\Gamma_\mathrm{N}} \delta u \, q^\mathrm{p} \, \mathrm{d}\Gamma +
+\int_\Omega \delta u \, b \, \mathrm{d}\Omega \quad \forall \, \delta u \in \mathbb{T}
 ```
 
-where $\mathbb{U}, \mathbb{U}^0$ are function spaces with sufficiently regular functions.
-It can be shown that the solution to the weak form is identical to the solution to the
-strong format.
+where $\mathbb{U}, \mathbb{T}$ are suitable function spaces with sufficiently regular
+functions. Under very general assumptions it can be shown that the solution to the weak
+form is identical to the solution to the strong form.
+
+## Finite Element Approximation
+
+TODO: Image of geometric discretization with triangles.
 
-## FE-approximation
+Using the finite element method to solve partial differential equations is usually
+preceded with the construction of a discretization of the domain $\Omega$ into a finite
+set of *elements* or *cells*. We call this geometric discretization *grid* (or *mesh*)
+and denote it with $\Omega_h$. In this example the corners of the triangles are called
+*nodes*.
 
-We now introduce the finite element approximation $u_h \approx u$ as a sum of shape
-functions, $N_i$ and nodal values, $a_i$. We approximate the test function in the same way
-(known as the Galerkin method):
+Next we introduce the finite element approximation $u_\mathrm{h} \approx u$ as a sum of N nodal
+*shape functions*, where we denote each of these function by $\phi_i$ and the corresponding *nodal
+values* $\hat{u}_i$. Depending on the community, shape functions are also called *trial functions*.
```

Comment:
```suggestion
*shape functions*, where we denote each of these function by $\phi_i$ and the corresponding *nodal values*
$\hat{u}_i$. Note that *shape functions* are sometimes referred to as *base functions* or *trial functions*, and instead of $\phi_i$ they are sometimes denoted $N_i$.
```